src/tree/localProject/LocalProjectTreeItem.ts

```diff
@@ -62,14 +63,27 @@ export class LocalProjectTreeItem extends LocalProjectTreeItemBase implements Di
         this._localFunctionsTreeItem = new LocalFunctionsTreeItem(this);
     }
 
-    public async getHostUrl(context: IActionContext): Promise<string> {
+    public async getHostRequest(context: IActionContext): Promise<FuncHostRequest> {
         let port = runningFuncPortMap.get(this.workspaceFolder);
         if (!port) {
             const funcTask: Task | undefined = (await tasks.fetchTasks()).find(t => t.scope === this.workspaceFolder && isFuncHostTask(t));
             port = await getFuncPortFromTaskOrProject(context, funcTask, this.effectiveProjectPath);
         }
 
-        return `http://localhost:${port}`;
+        const url = `http://localhost:${port}`;
+        try {
+            await requestUtils.sendRequestWithExtTimeout(context, { url: url, method: 'GET' });
+        } catch {
+            try {
+                const httpsUrl = url.replace('http', 'https');
+                await requestUtils.sendRequestWithExtTimeout(context, { url: httpsUrl, method: 'GET', rejectUnauthorized: false });
+                return { url: httpsUrl, rejectUnauthorized: false };
```

Comment:
I think adding a comment mentioning to update this later would be a good idea if this is not an optimal approach in terms of security.

Reply:
I don't plan to update this later. The reason I'm okay with it is because it's specifically for the `localhost` case, meaning it's only reaching a url on the user's machine. This is similar to how the Databases extension sets it to false for the emulator, which is also using localhost: https://github.com/microsoft/vscode-cosmosdb/blob/cb70ebec689a3217709546b258497c9e2ad1afa7/src/docdb/getCosmosClient.ts#L17